src/Sql/NotificationCenter/dbo/Tables/NotificationStatus.sql

```diff
@@ -5,5 +5,6 @@ CREATE TABLE [dbo].[NotificationStatus]
     [ReadDate] DATETIME2 (7) NULL,
     [DeletedDate] DATETIME2 (7) NULL,
     CONSTRAINT [PK_NotificationStatus] PRIMARY KEY CLUSTERED ([NotificationId] ASC, [UserId] ASC),
-    CONSTRAINT [FK_NotificationStatus_User] FOREIGN KEY ([UserId]) REFERENCES [dbo].[User] ([Id])
+    CONSTRAINT [FK_NotificationStatus_Notification] FOREIGN KEY ([NotificationId]) REFERENCES [dbo].[Notification] ([Id]) ON DELETE CASCADE,
+    CONSTRAINT [FK_NotificationStatus_User] FOREIGN KEY ([UserId]) REFERENCES [dbo].[User] ([Id]) ON DELETE NO ACTION 
```

Comment:
This `ON DELETE CASCADE` creates cycles and is unecessary, since the cascade flow is as follows: `delete user -> cascade delete notification > cascade delete notification status`

Reply:
Implement application-level cascading:
Handle the deletion of related records in your application code rather than relying on database-level cascades

Reply:
@rkac-bw the cascade pattern is all over the place for account deletion though, so the intention is to just keep going with that for now.

Reply:
Well it may be better to pull it back because SQL Server is preventing you from adding another cascade delete from User to NotificationStatus due to multiple cascade paths

Reply:
These are multipath deletes to same record:
User -> Notification -> NotificationStatus
User -> NotificationStatus


Deleting a User could delete NotificationStatus directly.
It could also delete Notification, which would then cascade to delete NotificationStatus.

Not allowed in SQ Server< Notification had ON DELETE CASCADE for UserId
NotificationStatus had ON DELETE CASCADE for NotificationId
You wanted to add ON DELETE CASCADE for UserId on NotificationStatus

Options:
- Switch to handling cascade deletes in code (my preference)
- Use triggers on user for on delete
- Use ON DELETE SET NULL and clean up process later (we should have a clean up process to clean up old and stale records regardless)
- Redesign the database schema to avoid multiple paths.

Reply:
Still not sure why we would delete the notification if there is no user to request it, and in the denormalized view would just be flagged as global but have the userid for receiving, Creating a  record for every user when it is global is an interesting issue in both scenarios.

Reply:
Alright, after some background chats and research I feel conflicted about what our strategy really is with cascades. We do need to deal with deletions, but I see us using them in conjunction with logic in the https://github.com/bitwarden/server/blob/main/src/Sql/dbo/Stored%20Procedures/User_DeleteById.sql proc. At the end of the day here and given how these optional FKs operate, let's just invest in the proc.

Reply:
Chatting about this and it looks like it would be best to have deletes caused by user to be run from here:

https://github.com/bitwarden/server/blob/main/src/Sql/dbo/Stored%20Procedures/User_DeleteById.sql

Reply:
Fixed in f1844e3